git-hg/updatetags8.sh

```diff
@@ -0,0 +1,134 @@
+[ $GIT_MAJOR_VERSION -eq 2 -a $GIT_MINOR_VERSION -lt 16 ] && echo I need git version 2.16 or later and you have $GIT_VERSION && exit 1
+# https://raw.githubusercontent.com/felipec/git-remote-hg/master/git-remote-hg
+if ! which git-remote-hg 2>/dev/null; then
+  echo I need git-remote-hg and could not find it
+  echo Get it from http://raw.githubusercontent.com/felipec/git-remote-hg/master/git-remote-hg
+  exit 1
+fi
+
+# Clone current AdoptOpenJDK repo
+cd $WORKSPACE/$GITHUB_REPO
+echo "Clone current $GITHUB_REPO"
+git clone $GITHUB_PROJECT/$GITHUB_REPO.git
+cd $GITHUB_REPO
+git fetch --tags
+OLDTAG=$(git describe --abbrev=0 --tags) 
+echo "Current openjdk level is $OLDTAG"
+
+# Clone current openjdk from Mercurial
+echo "Get base openjdk repository" 
+cd $WORKSPACE/openjdk/mirror
+git init
+git clone --bare "hg::http://hg.openjdk.java.net/$OPENJDK_VERSION/$OPENJDK_VERSION"
+
+cd "$OPENJDK_VERSION.git"
+
+git filter-branch -f --index-filter 'git rm -r -f -q --cached --ignore-unmatch .hg .hgignore .hgtags get_source.sh' --prune-empty --tag-name-filter cat -- --all
+
+cd ..
+git pull "$OPENJDK_VERSION"
+git fetch --tags "$OPENJDK_VERSION"
+# NEWTAG=$(git describe --abbrev=0 --tags)
+for NEWTAG in $TAGS ; do
+  cd $WORKSPACE/openjdk/mirror
+  git reset --hard $NEWTAG
+
+  echo $(date +%T): "Updating master branch for $NEWTAG"
+  cd $WORKSPACE/$GITHUB_REPO/$GITHUB_REPO
+  git checkout master
+  git fetch $WORKSPACE/openjdk/mirror
+  git merge --allow-unrelated-histories -m "Merge base $NEWTAG" FETCH_HEAD
+
+  for module in "${MODULES[@]}"
+    do
+      if [ ! -d "$WORKSPACE/openjdk/$module" ]; then
+        rm -rf "$WORKSPACE/openjdk/$module"
+        mkdir "$WORKSPACE/openjdk/$module"
+        cd $WORKSPACE/openjdk/$module
+        git init
+        echo $(date +%T):"Clone $module"
+        git clone --bare "hg::http://hg.openjdk.java.net/${OPENJDK_VERSION}/${OPENJDK_VERSION}/$module" || exit 1
+        echo $(date +%T): "GIT filter on $module"
+        cd $module.git
+        git filter-branch -f --index-filter "git rm -f -q --cached --ignore-unmatch .hgignore .hgtags && git ls-files -s | sed \"s|\t\\\"*|&$module/|\" | GIT_INDEX_FILE=\$GIT_INDEX_FILE.new git update-index --index-info && mv \"\$GIT_INDEX_FILE.new\" \"\$GIT_INDEX_FILE\"" --prune-empty --tag-name-filter cat -- --all
+        cd ..
+      fi
+      echo $(date +%T): : "GIT pull/reset on $module at $NEWTAG"
+      cd $WORKSPACE/openjdk/$module
+      git pull $module
+      git fetch --tags $module
+      git reset --hard $NEWTAG
+      cd $WORKSPACE/$GITHUB_REPO/$GITHUB_REPO 
+      git fetch $WORKSPACE/openjdk/$module
+      echo $(date +%T): "GIT filter on $module"
+      if ! git merge --allow-unrelated-histories -m "Merge $module at $NEWTAG" FETCH_HEAD; then
+         echo Please resolve the conflicts above in $WORKSPACE/$GITHUB_REPO/$GITHUB_REPO, then press return to continue
```

Comment:
This could hang the script during an automated run?  Might need to bomb out instead

Reply:
It certainly would.

Reply:
OK, probably needs to exit then